### PR TITLE
Fix config map options validation

### PR DESCRIPTION
**Pull request by pippolo84**
The validation of the config options loaded from the ConfigMap was carried out using the `pflag` package.
This is different from what Viper does when it tries to convert each string value in the appropriate config option type. Internally, Viper uses the `cast` package to convert the string in an appropriate value of the specific option type.

Using `pflag` was leading to false positives while checking the option values. One such example is this bug: https://github.com/cilium/cilium/issues/20173

To fix the validation, the APIs from the `cast` package are used, so to avoid (silent) misconfiguration while starting the cilium-agent or the cilium-operator.

Related to #20282

### CODE REVIEW

File: install/kubernetes/cilium/values.yaml
Comment (ldelossa) [inline install/kubernetes/cilium/values.yaml:33]: What occurs when this is empty? It doesn't get the zero value of "0" when parsed into Go structs?

**Review by ldelossa — APPROVED** [review]
Overall, looks good to me. Question about zero value in helm template.

File: install/kubernetes/cilium/values.yaml
Comment (pippolo84) [reply install/kubernetes/cilium/values.yaml:33]: When using the ConfigMap, the `cluster-id` option is loaded from the relevant file. Without the `0`, the file is empty, so Viper (silently) fails to convert an empty string to an int, here:

https://github.com/cilium/cilium/blob/c3cd7da63885a320435a63a70cd7d9c7c5971abd/pkg/option/config.go#L2763

Prior to this PR, nothing too bad was happening: the default value for `cluster-id` is `0` and this is also the Go zero value for the `int` type. Thus, viper was failing silently, but the `c.ClusterID` was left with a `0` anyway.

But the proposed changes strengthen the validation process on option values. Without specifying a correct value, we would end up with this error:

`option cluster-id: unable to cast "" of type string to int64`

Comment (ldelossa) [reply install/kubernetes/cilium/values.yaml:33]: Okay I see, and I agree with your reasoning. LGTM.

File: pkg/option/config.go
Comment (sayboras) [inline pkg/option/config.go:3826]: oh I thought we removed the flag name in error message, but it's available in err itself 💯

**Review by sayboras — APPROVED** [review]
LGTM 💯

**Review by gandro — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Review by twpayne — APPROVED** [review]
Very nice, thank you :)

**Review by jibi — APPROVED** [review]

**Comment by pippolo84** [comment]
> (do we want to mark it for backport like #20282?)

Probably not, because we have changed the requirements for the `cluster.id` helm value, but I leave it up to @gandro to confirm that.

**Comment by gandro** [comment]
> > (do we want to mark it for backport like #20282?)
> 
> Probably not, because we have changed the requirements for the `cluster.id` helm value, but I leave it up to @gandro to confirm that.

Yeah, the `cluster.id` change (which is technically breaking) makes me a bit nervous to backport to any already released version. I would say let's hold off the backport to v1.11 and older, unless someone actually hits bug in the wild.

**Pull request merged**
